Subject: [PATCH] [MC937][TaskID]120725[Description]camera not working[solution]release gpio73\gpio74[owner]wangzhiwen

---

diff --git a/937/pvt/yupik-camera-937-pvt.dtsi b/937/pvt/yupik-camera-937-pvt.dtsi
index b6becc4..4369524 100644
--- a/937/pvt/yupik-camera-937-pvt.dtsi
+++ b/937/pvt/yupik-camera-937-pvt.dtsi
@@ -754,18 +754,15 @@
 		src-clock-name = "cci_1_clk_src";
 		clock-cntl-level = "lowsvs";
 		clock-rates = <37500000 0>;
+                //pinctrl-0 = <&cci3_active>;pinctrl-1 = <&cci3_suspend>; cam_default is active.
 		pinctrl-names = "cam_default", "cam_suspend";
-		pinctrl-0 = <&cci2_active &cci3_active>;
-		pinctrl-1 = <&cci2_suspend &cci3_suspend>;
-		gpios = <&tlmm 73 0>,
-			<&tlmm 74 0>,
-			<&tlmm 75 0>,
+		pinctrl-0 = <&cci3_active>;
+		pinctrl-1 = <&cci3_suspend>;
+		gpios = <&tlmm 75 0>,
 			<&tlmm 76 0>;
-		gpio-req-tbl-num = <0 1 2 3>;
-		gpio-req-tbl-flags = <1 1 1 1>;
-		gpio-req-tbl-label = "CCI_I2C_DATA2",
-					"CCI_I2C_CLK2",
-					"CCI_I2C_DATA3",
+		gpio-req-tbl-num = <0 1>;
+		gpio-req-tbl-flags = <1 1>;
+		gpio-req-tbl-label = 	"CCI_I2C_DATA3",
 					"CCI_I2C_CLK3";
 
 		i2c_freq_100Khz_cci1: qcom,i2c_standard_mode {
